src/main/java/radiounit/EricssonLteRadioUnitReceiver.java

```diff
@@ -13,60 +15,193 @@
  */
 public class EricssonLteRadioUnitReceiver implements RadioUnitReceiver {
 
-	private List<Carrier> carriers;
+	/*
+	 * This is the critical resource of a radio.
+	 * 
+	 * The "volatile" keyword ensures that this resource is not cached, and is retrieved directly from
+	 * 	memory every time it is accessed.
+	 * 
+	 * The three "-Internal" methods below are all synchronized and have exclusive access to this resource. 
+	 */
+	private volatile List<Carrier> carriers;
+	
+	public EricssonLteRadioUnitReceiver()
+	{
+		this.carriers = new ArrayList<Carrier>();
+	}
 
 	public void setupEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] setupEricssonLte");
 	}
 
 	public void activateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] activateEricssonLte");
 	}
 
 	public void deactivateEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] deactivateEricssonLte");
 	}
 
 	public void releaseEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] releaseEricssonLte");
 	}
 
-	public void setupCarrierEricssonLte(Carrier carrrier) {
-		// TODO Auto-generated method stub
-
+	public void setupCarrierEricssonLte(Carrier carrier) {
+		System.out.println("[EricssonLteRadioUnitReceiver] setupCarrierEricssonLte: " + carrier);
+		
+		// Validate carrier
+		FrequencyBand band = carrier.getCarrierFrequencies();
+		boolean isLte = false;
+		
+		for (LteFrequencyBand lteFreq : LteFrequencyBand.values())
+		{
+			if (band == lteFreq)
+			{
+				isLte = true;
+				break;
+			}
+		}
+
+		if (!isLte)
+		{
+			System.err.println("Cannot add WCDMA carrier to LTE radio!");
+			return;
+		}
+		
+		if (getCarrierById(carrier.getCarrierId()) != null)
+		{
+			System.err.println("Carrier with id " + carrier.getCarrierId() + " already exists!");
+			return;
+		}
+		
+		addCarrierInternal(carrier);
 	}
 
 	public void signalScalingEricssonLte() {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] signalScalingEricssonLte");
 	}
 
 	public void modifyCarrierEricssonLte(Integer carrierId, FrequencyBand frequencyBand) {
-		// TODO Auto-generated method stub
-
+		System.out.println("[EricssonLteRadioUnitReceiver] modifyCarrierEricssonLte: " + carrierId + ", " + frequencyBand.getBand());
+		
+		int idx = getCarrierIdxById(carrierId);
+		
+		if (idx == -1)
```

Comment:
What does -1 represent? Is it the default value? 

Would this be considered a magic number?

Reply:
Ah, now I understand your question from Teams!

The -1 represents that there is no carrier in the list of carriers with that index - so kinda like a default value.

This is a very common convention used places - when you're looking for an object in a list, you get the index of that value, and then access it. If it doesn't exist, you return a known value (usually -1).

Technically speaking, this is a magic number, however it's a common enough convention that it isn't typically extracted to a constant. That being said, I can do that to make it more clear.

Reply:
Ah, okay. If it's a common convention, that makes sense.